lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/releases/ReleaseProviderException.scala

```diff
@@ -1,12 +1,35 @@
 package org.enso.runtimeversionmanager.releases
 
-/** Indicates a release provider failure.
-  */
-case class ReleaseProviderException(message: String, cause: Throwable = null)
+/** Indicates a release provider failure. */
+sealed class ReleaseProviderException(message: String, cause: Throwable)
     extends RuntimeException(message, cause) {
 
-  /** @inheritdoc
-    */
+  /** @inheritdoc */
   override def toString: String =
     s"A problem occurred when trying to find the release: $message"
 }
+
+object ReleaseProviderException {
+
+  /** Creates a release provider exception with a given message and optional
+    * cause.
+    */
+  def apply(
+    message: String,
+    cause: Throwable = null
+  ): ReleaseProviderException = new ReleaseProviderException(message, cause)
+}
+
+/** Indicates that the specific release could not be found.
+  *
+  * @param tag the tag of the release
+  * @param cause (optional) an exception that has caused this error
+  */
+case class ReleaseNotFound(
+  tag: String,
+  message: Option[String] = None,
+  cause: Throwable        = null
+) extends ReleaseProviderException(
+      message.getOrElse(s"Cannot find release `$tag`."),
```

Comment:
If we hit this case will it check the fallback as well?

Reply:
Yes, the provider with fallback tries the fallback on any error.
```scala
  /** @inheritdoc
    */
  override def releaseForTag(tag: String): Try[Release] =
    baseProvider.releaseForTag(tag).recoverWith { error =>
      if (fallbackProvider.isEnabled)
        fallbackProvider.releaseForTag(tag)
      else Failure(error)
    }
```

Because we cannot really be sure if the 404 not found is due to nightly being removed or the Github service acting weirdly.